### PR TITLE
Add stubs for MSAL token getters in node `WorkloadIdentityCredential` tests

### PR DESCRIPTION
**Pull request by dgetu**

### CODE REVIEW

File: sdk/identity/identity/recordings/node/workloadidentitycredential/recording_authenticates_with_defaultazure_credential.json
Comment (KarishmaGhiya) [inline sdk/identity/identity/recordings/node/workloadidentitycredential/recording_authenticates_with_defaultazure_credential.json:1]: Can you please add the recordings back?

Comment (dgetu) [reply sdk/identity/identity/recordings/node/workloadidentitycredential/recording_authenticates_with_defaultazure_credential.json:1]: Done!

File: sdk/identity/identity/test/msalTestUtils.ts [hunk 1]
```diff
@@ -76,152 +81,202 @@ export const openIdConfigurationResponse: Record<string, string | string[] | boo
...
+export async function msalNodeTestSetup(options: {
+  testContext?: Mocha.Test;
+  playbackClientId?: string;
+}): Promise<{
+  cleanup: MsalTestCleanup;
+  recorder: Recorder;
+  sandbox: Sinon.SinonSandbox;
+}>;
+
+export async function msalNodeTestSetup(options: { stubbedToken: AuthenticationResult }): Promise<{
```
Comment (KarishmaGhiya) [inline sdk/identity/identity/test/msalTestUtils.ts]: what about these 2 overloads? Will this be supported by Typescript?

export async function msalNodeTestSetup( testContext?: Mocha.Test,  playbackClientId = "azure_client_id")
export async function msalNodeTestSetup(stubbedToken: AuthenticationResult)

Comment (dgetu) [reply sdk/identity/identity/test/msalTestUtils.ts]: Yes, I've just changed the signature to match.

**Pull request closed without merging**